r2/service.go

```diff
@@ -122,6 +125,62 @@ type r2Handler struct {
 	auth  auth.HTTPAuthService
 }
 
+type routeFunc func(s *service, r *http.Request) (data interface{}, err error)
+
+type serviceMetrics struct {
+	sync.RWMutex
+	scope tally.Scope
+
+	// Endpoints mapped to metrics (endpoint = path + method + status).
+	metrics map[string]serviceMetric
+}
+
+type serviceMetric struct {
+	timer   tally.Timer
+	counter tally.Counter
+}
+
+// TODO(sandusky) Wrap each route with this function.
+func handleRoute(rf routeFunc, s *service, r *http.Request, namespace string) (interface{}, error) {
+	start := time.Now()
```

Comment:
I'd pass in a clock.Options (see `m3db/m3x/clock`) to the service constructor, store the `NowFn` in the service, and use the `NowFn` here so you can easily provide a different `nowFn` if needed (e.g., in tests).